src/checks/eslint.ts

```diff
@@ -1,35 +1,33 @@
+import { ESLint } from 'eslint';
 import { execaCommand } from 'execa';
-import normalizeNewline from '../util/normalizeNewline';
+
+const errorKey = 'errorCount' as const;
+const warningKey = 'warningCount' as const;
 
 export const title = 'ESLint';
 
 export default async function eslintCheck() {
   try {
-    await execaCommand('pnpm eslint . --max-warnings 0  --format compact');
+    await execaCommand('pnpm eslint . --max-warnings 0  --format json');
   } catch (error) {
     const { stdout } = error as { stdout: string };
-    const lines = stdout.split('\n');
 
     // If no ESLint problems detected, throw the error
-    if (!/^\d+ problems?$/.test(lines[lines.length - 2]!)) {
+    if (!new RegExp(`"${errorKey}":|"${warningKey}":`).test(stdout)) {
       throw error;
     }
 
     throw new Error(
       `ESLint problems found in the following files:
-        ${normalizeNewline(stdout)
-          .split('\n')
-          // Match lines starting with slashes (macOS, Linux) or drive letters (Windows)
-          .filter((line) => /^(\/|[A-Z]:\\)/.test(line))
-          // Strip out the filename
-          .map((line) => line.match(/^(([A-Z]:)?[^:]+):/)?.[1])
-          // Remove duplicate filenames
-          .reduce((linesWithoutDuplicates: string[], line) => {
-            if (line && !linesWithoutDuplicates.includes(line)) {
-              linesWithoutDuplicates.push(line);
-            }
-            return linesWithoutDuplicates;
-          }, [])
+        ${(JSON.parse(stdout) as ESLint.LintResult[])
+          .filter(
+            (eslintResult) =>
+              !(eslintResult[errorKey] === 0 && eslintResult[warningKey] === 0),
```

Comment:
how would this be the case? what is a a `LintResult` which has both `errorCount` and `warningCount` both of `0`? does this happen in reality?

if it does, then this probably needs a comment describing when this is the case